### PR TITLE
Replies

### PR DESCRIPTION
**Pull request by ArgoNavis1763**
Edit and delete functionality working for replies

### CODE REVIEW

File: cypress/integration/comments-section/actioning-comments.spec.js [hunk 1]
```diff
@@ -190,6 +188,47 @@ describe('viewing comments', () => {
...
+    cy.get('form').submit();
+    cy.get('[id="replies-container"]').contains('hello');
+  });
+
+  it('delete modal should appear when delete clicked on own reply', () => {
+    cy.get('[id="reply-delete-button"]').click();
+    cy.get('[id="delete-modal"]').should('exist');
+  });
+
+  it.only('should delete own reply when delete button in modal is clicked', () => {
```
Comment (TomGroombridge) [inline cypress/integration/comments-section/actioning-comments.spec.js]: remove .only

File: src/components/DeleteReplyModal.jsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import { Dialog, Transition } from '@headlessui/react';
+import { ExclamationIcon } from '@heroicons/react/outline';
+import React, { Fragment, useRef } from 'react';
+
+const DeleteReplyModal = (props) => {
+  const handleDelete = () => {
+    console.log('props.comments', props.comments);
```
Comment (TomGroombridge) [inline src/components/DeleteReplyModal.jsx]: remove console

File: src/components/DeleteReplyModal.jsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    const newComments = props.comments.map((comment) => {
+      if (comment.id !== props.commentId) {
+        return comment;
+      } else if (comment.id === props.commentId) {
```
Comment (TomGroombridge) [inline src/components/DeleteReplyModal.jsx]: this can just be else

File: src/components/comment.jsx [hunk 4]
```diff
@@ -100,16 +102,18 @@ const Comment = (props) => {
...
-              className="border-2 border-purple-900 text-purple-900 p-1 m-1 flex items-center"
+              className="text-purple-900 bg-white p-1 mx-1 justify-between flex items-center w-[76px]"
               id="reply-button"
               onClick={() => {
                 setReplyClicked(true);
               }}
             >
-              <p>Reply</p>
-              <img src="/icons/icon-reply.svg" />
+              {/* <div className="flex items-center mx-1 w-[70px] justify-between"> */}
```
Comment (TomGroombridge) [inline src/components/comment.jsx]: remove comment

File: src/components/reply.jsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+          onClick={handleReplyMinus}
+          className="p-1 m-1"
+        >
+          -
+        </button>
+      </div>
+      <div
+        // className="bg-slate-200 m-1 p-1 text-sm w-[500px]"
+        id={`reply-${props.reply.id}`}
+        key={props.index}
```
Comment (TomGroombridge) [inline src/components/reply.jsx]: why is this key here?

File: src/components/comment.jsx [hunk 6]
```diff
@@ -136,14 +140,17 @@ const Comment = (props) => {
...
-                key={index}
-              >
-                <p>{reply.user.username}</p>
-                <p>{reply.content}</p>
-              </div>
+              <Reply
+                reply={reply}
+                index={index}
+                replies={props.comment.replies}
+                id={reply.id}
```
Comment (TomGroombridge) [inline src/components/comment.jsx]: can we name this prop better?

File: src/components/comment.jsx [hunk 7]
```diff
@@ -3,11 +3,13 @@ import Edit from './edit';
 import Delete from './delete';
 import { v4 as uuidv4 } from 'uuid';
 import Comments from '../pages/Comments';
+import Reply from './reply';
 
 const Comment = (props) => {
   const [replyClicked, setReplyClicked] = useState(false);
   const [replyValue, setReplyValue] = useState('');
   const [comment, setComment] = useState(props.comment);
+  const [replies, setReplies] = useState(props.comment.replies);
```
Comment (TomGroombridge) [inline src/components/comment.jsx:11]: is this needed? the `replies` seems to never be used?

File: src/components/reply.jsx [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    }
+    const editedReplies = props.replies.map((reply, index) => {
+      if (props.id === reply.id) {
+        reply.content = content;
+      }
+      return reply;
+    });
+
+    // props.replies = editedReplies;
+    props.setReplies(editedReplies);
```
Comment (TomGroombridge) [inline src/components/reply.jsx]: I'm not sure this is used?

File: src/components/comment.jsx [hunk 9]
```diff
@@ -3,11 +3,13 @@ import Edit from './edit';
 import Delete from './delete';
 import { v4 as uuidv4 } from 'uuid';
 import Comments from '../pages/Comments';
+import Reply from './reply';
 
 const Comment = (props) => {
   const [replyClicked, setReplyClicked] = useState(false);
   const [replyValue, setReplyValue] = useState('');
   const [comment, setComment] = useState(props.comment);
```
Comment (TomGroombridge) [inline src/components/comment.jsx]: again does this need to be state?

File: src/components/reply.jsx [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+  };
+
+  const handleReplyPlus = () => {
+    const newReplies = props.replies.map((reply, index) => {
+      if (reply.id === props.reply.id) {
+        reply.score += 1;
+      }
+      return reply;
+    });
+    // props.replies = newReplies;
```
Comment (TomGroombridge) [inline src/components/reply.jsx]: remove comment

**Review by TomGroombridge — COMMENTED** [review]
few comments, looks great though

File: cypress/integration/comments-section/actioning-comments.spec.js (hunk 1 above)
Comment (ArgoNavis1763) [reply cypress/integration/comments-section/actioning-comments.spec.js]: done

File: src/components/DeleteReplyModal.jsx (hunk 2 above)
Comment (ArgoNavis1763) [reply src/components/DeleteReplyModal.jsx]: done

File: src/components/DeleteReplyModal.jsx (hunk 3 above)
Comment (ArgoNavis1763) [reply src/components/DeleteReplyModal.jsx]: done

File: src/components/comment.jsx (hunk 4 above)
Comment (ArgoNavis1763) [reply src/components/comment.jsx]: done

File: src/components/reply.jsx (hunk 5 above)
Comment (ArgoNavis1763) [reply src/components/reply.jsx]: removed

File: src/components/comment.jsx (hunk 6 above)
Comment (ArgoNavis1763) [reply src/components/comment.jsx]: done

File: src/components/reply.jsx (hunk 8 above)
Comment (ArgoNavis1763) [reply src/components/reply.jsx]: removed

File: src/components/reply.jsx (hunk 10 above)
Comment (ArgoNavis1763) [reply src/components/reply.jsx]: removed

File: src/components/comment.jsx (hunk 9 above)
Comment (ArgoNavis1763) [reply src/components/comment.jsx]: Did you mean line 6 or line 12 for this one?

File: src/components/comment.jsx (hunk 7 above)
Comment (ArgoNavis1763) [reply src/components/comment.jsx:11]: setReplies is used though so we'll need to create a separate function for it

Comment (TomGroombridge) [reply src/components/comment.jsx:11]: if you use `setReplies` but never use `replies` then you're setting something that will never be used.

**Pull request merged**